src/librepgp/stream-common.cpp

```diff
@@ -305,30 +305,35 @@ src_skip_eol(pgp_source_t *src)
 bool
 src_peek_line(pgp_source_t *src, char *buf, size_t len, size_t *readres)
 {
-    size_t clen = 0;
-    size_t read;
-
-    /* we need some place for \0 */
-    len--;
+    size_t scan_pos = 0;
+    size_t inc = 64;
+    len = len - 1;
 
     do {
-        read = clen + 64 > len ? len - clen : 64;
-        if (!src_peek(src, buf + clen, read, &read) || !read) {
+        size_t remaining = len - scan_pos;
+        size_t to_peek = scan_pos + (remaining > inc ? inc : remaining);
+        inc = inc * 2;
```

Comment:
The idea here is to peek data in chunks, to not read too much at once, which could affect performance. While most lines would be shorter than 64 bytes, we should not peek the whole 1024 at once. Multiplying by 2 is also part of performance strategy - it is to do `log(len)` iterations instead of linear len/64.

Reply:
@ni4 I understand that you're trying to create a sequence of `to_peek` = 64, 192, 448, 960 ....
It's just evident that sequence will be the same regardless of `len`, `len` only serves as a ceiling, so we could program it in a simpler manner
```
size_t x = scan_pos + inc;
size_t to_peek = (x > len) ? x : len;
inc = inc * 2;
``` 
This will replicate your sequence but in simpler expressions.
Or we can do almost the same sequence:
```
size_t to_peek = len > inc ? inc : len;
inc = inc * 3;
```
It will be 64, 192, 576, 1728 ...